src/replica/replica_disk_migrator.cpp

```diff
@@ -148,12 +152,140 @@ bool replica_disk_migrator::check_migration_args(const replica_disk_migrate_requ
     return true;
 }
 
-// TODO(jiashuo1)
 // THREAD_POOL_REPLICATION_LONG
-void replica_disk_migrator::migrate_replica(const replica_disk_migrate_request &req) {}
+void replica_disk_migrator::do_disk_migrate_replica(const replica_disk_migrate_request &req)
+{
+    if (status() != disk_migration_status::MOVING) {
+        derror_replica("disk migration(origin={}, target={}), err = Invalid migration status({})",
+                       req.origin_disk,
+                       req.target_disk,
+                       enum_to_string(status()));
+        reset_status();
+        return;
+    }
+
+    if (init_target_dir(req) && migrate_replica_checkpoint(req) && migrate_replica_app_info(req)) {
+        _status = disk_migration_status::MOVED;
+        ddebug_replica("disk migration(origin={}, target={}), update status from {}=>{}, ready to "
+                       "close origin replica({})",
+                       req.origin_disk,
+                       req.target_disk,
+                       enum_to_string(disk_migration_status::MOVING),
+                       enum_to_string(status()),
+                       _replica->dir());
+
+        close_current_replica();
+    }
+}
+
+// THREAD_POOL_REPLICATION_LONG
+bool replica_disk_migrator::init_target_dir(const replica_disk_migrate_request &req)
+{
+    // replica_dir: /root/origin_disk_tag/gpid.app_type
+    std::string replica_dir = _replica->dir();
+    // using origin dir to init new dir
+    boost::replace_first(replica_dir, req.origin_disk, req.target_disk);
+    if (utils::filesystem::directory_exists(replica_dir)) {
+        derror_replica("migration target replica dir({}) has existed", replica_dir);
+        reset_status();
+        return false;
+    }
+
+    // _target_replica_dir = /root/target_disk_tag/gpid.app_type.disk.balance.tmp, it will update to
+    // /root/target_disk_tag/gpid.app_type in replica_disk_migrator::update_replica_dir finally
+    _target_replica_dir = fmt::format("{}{}", replica_dir, kReplicaDirTempSuffix);
+    if (utils::filesystem::directory_exists(_target_replica_dir)) {
+        dwarn_replica("disk migration(origin={}, target={}) target replica dir({}) has existed, it "
+                      "will be deleted",
+                      req.origin_disk,
+                      req.target_disk,
+                      _target_replica_dir);
+        utils::filesystem::remove_path(_target_replica_dir);
+    }
+
+    //  _target_replica_data_dir = /root/gpid.app_type.disk.balance.tmp/data/rdb, it will update to
+    //  /root/target/gpid.app_type/data/rdb in replica_disk_migrator::update_replica_dir finally
+    _target_data_dir = utils::filesystem::path_combine(_target_replica_dir, kDataDirFolder);
```

Comment:
Are `_target_replica_dir = fmt::format("{}{}", replica_dir, kReplicaDirTempSuffix);` and `_target_data_dir = utils::filesystem::path_combine(_target_replica_dir, kDataDirFolder);` the same?

Reply:
filesystem::path_combine("/abc", "file") => /abc/file, but fommat("/abc", "file")=>/abcfile